src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -1237,6 +1244,234 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_SetupSrpHost(co
     return error;
 }
 
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD_DNS_CLIENT
+template <class ImplClass>
+void GenericThreadStackManagerImpl_OpenThread<ImplClass>::FromOtDnsResponseToMdnsData(otDnsServiceInfo & serviceInfo,
+                                                                                      const char * serviceType,
+                                                                                      chip::Mdns::MdnsService & mdnsService,
+                                                                                      DnsServiceTxtEntries & serviceTxtEntries)
+{
+    char protocol[chip::Mdns::kMdnsProtocolTextMaxSize + 1];
+
+    // Extract from the <hostname>.<domain-name>. the <hostname> part.
+    size_t substringSize = strlen(serviceInfo.mHostNameBuffer) - strlen(strchr(serviceInfo.mHostNameBuffer, '.'));
```

Comment:
Why isn't this just:
```
  substringSize = strchr(serviceInfo.mHostNameBuffer, '.') - serviceInfo.mHostNameBuffer;
```
or something along those lines?

Also, both versions (the one in the PR and the one above) seem to break down if `'.'` is not present in the string at all, right?

Similar questions for other case below.

Reply:
Both works, but your way is simpler I guess. If it comes to the breaking down if `.` is not present - I believe OpenThreads method will not return anything not following this pattern, as it will indicate error first. Nevertheless your point is of course valid and we should not blindly trust it. That's why I changed method API to return CHIP_ERROR instead of void and verify in strings contain dots.